02_classify_mutations/nbconverted/compare_filtering.py

```diff
@@ -0,0 +1,300 @@
+old_results_df.head()
+
+
+# In[12]:
+
+
+old_all_results_df = au.compare_all_data_types(old_results_df,
+                                               SIG_ALPHA,
+                                               metric='aupr')
+
+old_all_results_df.sort_values(by='p_value').head(10)
+
+
+# In[13]:
+
+
+overlap_genes = (
+    set(new_all_results_df.gene.unique())
+      .intersection(set(old_all_results_df.gene.unique()))
+)
+
+new_all_results_df = new_all_results_df[new_all_results_df.gene.isin(overlap_genes)]
+old_all_results_df = old_all_results_df[old_all_results_df.gene.isin(overlap_genes)]
+
+cols = ['gene', 'training_data', 'delta_mean']
+diff_df = (old_all_results_df
+  .loc[:, cols]
+  .merge(new_all_results_df.loc[:, cols],
+         left_on=['gene', 'training_data'],
+         right_on=['gene', 'training_data'])
+)
+
+diff_df.rename(
+    columns={'delta_mean_x': 'delta_mean_old',
+             'delta_mean_y': 'delta_mean_new'},
+    inplace=True
+)
+diff_df['old_vs_new'] = diff_df.delta_mean_old - diff_df.delta_mean_new
+diff_df.head()
+
+
+# In[14]:
+
+
+sns.set({'figure.figsize': (12, 6)})
+
+sns.boxplot(data=diff_df, x='training_data', y='old_vs_new',
+            order=training_data_map.values())
+plt.xlabel('Training data')
+plt.ylabel('(Old - new) filtering scheme')
+
+
+# In[15]:
+
+
+metric = 'aupr'
+num_examples = 20
+
+top_df = (diff_df
+    .sort_values(by='old_vs_new', ascending=False)
+    .head(num_examples)
+)
+bottom_df = (diff_df
+    .sort_values(by='old_vs_new', ascending=False)
+    .tail(num_examples)
+)
+
+plot_df = pd.concat((top_df, bottom_df)).reset_index()
+
+sns.set({'figure.figsize': (30, 5)})
+sns.barplot(data=plot_df, x=plot_df.index, y='old_vs_new',
+            hue='training_data', dodge=False,
+            hue_order=training_data_map.values())
+
+def show_values_on_bars(ax):
+    for i in range(plot_df.shape[0]):
+        _x = i
+        _y = plot_df.loc[i, 'old_vs_new']
+        val = plot_df.loc[i, 'gene']
+        if _y > 0:
+            ax.text(_x, _y + 0.02, val, ha="center") 
+        else:
+            ax.text(_x, _y - 0.04, val, ha="center") 
+        
+show_values_on_bars(plt.gca())
+plt.gca().get_xaxis().set_visible(False)
+plt.ylabel('AUPR difference')
+plt.title('Performance difference between old and new filtering scheme', size=14)
+
+
+# The plots above show the difference for each gene between the "old" filtering scheme (per-cancer type) and the "new" filtering scheme (gene-level across all cancer types). A positive value for a gene means that gene's classifier performed better for the "old" filtering scheme and vice-versa for the "new" filtering scheme.
+# 
+# We can see that most genes have a positive old vs. new difference, indicating that filtering for each cancer type independently (the "old" approach) tends to lead to better classifier performance for most genes. This seems to hold across all the data types we looked at.
```

Comment:
It's a little weird to say "We can see that most genes have a positive old vs. new difference" right after a bar plot with equal numbers of classifiers in the positive and negative directions. I'd make it more obvious that the bar plot is looking at the tails, and if you want to you can also make a note that the most positive tail is bigger than the most negative tail.

Reply:
Good point! That line was mostly referring to the box plot (where the means are all greater than 0) but you're right that it would be clearer if I elaborate a bit.